web3/method.py

```diff
@@ -9,20 +9,21 @@
 )
 
 
-def star_apply(fn):
+def _munger_star_apply(fn):
     @functools.wraps(fn)
-    def inner(args):
-        return fn(*args)
+    def inner(module_args):
+        module, args = module_args
```

Comment:
The current API of having all the mungers return the `module` and then pass it onto the next seems problematic/error prone.  Instead of requiring the munger to both be accepted as an argument as well I think we should do the following.

1. only pass as argument.
2. use `curry` or `functools.partial` to set the `module` parameter on all the munger functions before passing the values through.

This would make the `map(star_apply, mungers)` into something like:

```python
map(lambda munger: star_apply(functools.partial(munger, module)), mungers)
```